classes/SpecUploadBase.php

```diff
@@ -984,8 +984,13 @@ protected function transferOccurrences(){
 					$insertCnt = $this->conn->affected_rows;
 					$warnCnt = $this->conn->warning_count;
 					if($warnCnt){
-						if(strpos($this->conn->get_warnings()->message,'UNIQUE_occurrenceID'))
-							$this->outputMsg('<li style="margin-left:10px"><span style="color:orange">WARNING</span>: '.$warnCnt.' records failed to load due to duplicate occurrenceID values which must be unique across all collections)</li>');
+						$this->outputMsg('<li style="margin-left:10px"><span style="color:orange">WARNING</span>: '.$warnCnt.' records failed to load:</li>');
+						$warning = $this->conn->get_warnings();
+						if ($warning) {
+							do {
+								$this->outputMsg('<li style="margin-left:10px">'.$warning->errno.': '.$warning->message.'</li>');
```

Comment:
Presenting raw error from the DB to the user could be a security risk.  Better to have logic that catches a specific problem.  If a general message is needed, point the user to their system admin / system logs.

Reply:
Add additional 'if, else' logic to handle error cases as needed with a generic message to check the logs.